warehouse/query-core/src/main/java/datawave/query/util/sortedset/FileSortedSet.java

```diff
@@ -573,79 +601,68 @@ public void clear() {
             handler.deleteFile();
             persisted = false;
         } else {
-            set.clear();
+            super.clear();
         }
     }
 
     @Override
     public Comparator<? super E> comparator() {
-        return set.comparator();
+        return super.comparator();
     }
 
     @Override
-    public SortedSet<E> subSet(E fromElement, E toElement) {
+    public RewritableSortedSet<E> subSet(E fromElement, E toElement) {
         return factory.newInstance(this, fromElement, toElement);
     }
 
     @Override
-    public SortedSet<E> headSet(E toElement) {
+    public RewritableSortedSet<E> headSet(E toElement) {
         return factory.newInstance(this, null, toElement);
     }
 
     @Override
-    public SortedSet<E> tailSet(E fromElement) {
+    public RewritableSortedSet<E> tailSet(E fromElement) {
         return factory.newInstance(this, fromElement, null);
     }
 
     @Override
     public E first() {
-        boolean gotFirst = false;
-        E first = null;
         if (persisted) {
             try (SortedSetInputStream<E> stream = getBoundedFileHandler().getInputStream(getStart(), getEnd())) {
-                first = stream.readObject();
-                gotFirst = true;
+                return stream.readObject();
             } catch (Exception e) {
                 QueryException qe = new QueryException(DatawaveErrorCode.FETCH_FIRST_ELEMENT_ERROR, e);
                 throw (new IllegalStateException(qe));
             }
-        } else if (!set.isEmpty()) {
-            first = set.first();
-            gotFirst = true;
-        }
-        if (!gotFirst) {
-            QueryException qe = new QueryException(DatawaveErrorCode.FETCH_FIRST_ELEMENT_ERROR);
-            throw (NoSuchElementException) (new NoSuchElementException().initCause(qe));
         } else {
-            return first;
+            return super.first();
         }
     }
 
     @Override
     public E last() {
-        boolean gotLast = false;
-        E last = null;
         if (persisted) {
+            boolean gotLast = false;
+            E last = null;
```

Comment:
I think these changes will go away once you merge integration in.

Reply:
Not sure what you meant here.  integration is merged in.

Reply:
On June 29th when I wrote this comment, integration hadn't been merged in and there were some changes missing here.